src/MDMAnimationTraits.h

```diff
@@ -0,0 +1,114 @@
+/*
+ Copyright 2017-present The Material Motion Authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <CoreGraphics/CoreGraphics.h>
+#import <Foundation/Foundation.h>
+
+#import "MDMRepetitionTraits.h"
+#import "MDMTimingCurve.h"
+
+/**
+ A generic representation of animation traits.
+ */
+@interface MDMAnimationTraits: NSObject
+
+/**
+ Initializes the instance with the provided duration and default iOS ease in/out cubic bezier.
+
+ @param duration The animation will occur over this length of time, in seconds.
+ */
+- (nonnull instancetype)initWithDuration:(NSTimeInterval)duration;
+
+/**
+ Initializes the instance with the provided duration, delay, and default iOS ease in/out cubic
+ bezier.
+
+ @param delay The amount of time, in seconds, to wait before starting the animation.
```

Comment:
Is any non-positive delay treated as "instantaneous" or would the animation be reversed?

Reply:
Negative delay would shift the animation backward in time. E.g if animating linearly from 0 to 1 over 10 second, a delay of -5 would start the animation at 0.5 and animate for 5 seconds before completing.

Reply:
And a negative duration would animate backwards until the duration is met?

Reply:
It appears that Core Animation will default to a duration of 0.25 when the duration is <= 0. Wonderful! https://github.com/material-motion/motion-animator-objc/issues/87 filed to track.